internal/setup/setup_test.go

```diff
@@ -1063,6 +1063,174 @@ var setupTests = []setupTest{{
+			},
+		},
+		Packages: map[string]*setup.Package{
+			"mypkg": {
+				Archive: "ubuntu",
+				Name:    "mypkg",
+				Path:    "slices/mydir/mypkg.yaml",
+				Slices: map[string]*setup.Slice{
+					"myslice": {
+						Package: "mypkg",
+						Name:    "myslice",
+						Contents: map[string]setup.PathInfo{
+							"/path/**": {Kind: "generate", Generate: "manifest"},
+						},
+					},
+				},
+			},
+			"foo": {
+				Archive: "ubuntu",
+				Name:    "foo",
+				Path:    "slices/mydir/foo.yaml",
+				Slices: map[string]*setup.Slice{
+					"bar": {
+						Package: "foo",
+						Name:    "bar",
+						Contents: map[string]setup.PathInfo{
+							"/foo/**": {Kind: "generate", Generate: "manifest"},
+						},
+					},
+				},
+			},
+		},
+	},
+	selslices: []setup.SliceKey{{"mypkg", "myslice"}},
+	selection: &setup.Selection{
+		Slices: []*setup.Slice{{
+			Package: "mypkg",
+			Name:    "myslice",
+			Contents: map[string]setup.PathInfo{
+				"/path/**": {Kind: "generate", Generate: "manifest"},
+			},
+		}},
+	},
+}, {
+	summary: "Can specify generate with bogus value but cannot SELECT those slices",
+	input: map[string]string{
+		"slices/mydir/mypkg.yaml": `
+			package: mypkg
+			slices:
+				myslice:
+					contents:
+						/path/**: {generate: "foo"}
+		`,
+	},
+	release: &setup.Release{
+		DefaultArchive: "ubuntu",
+
+		Archives: map[string]*setup.Archive{
+			"ubuntu": {
+				Name:       "ubuntu",
+				Version:    "22.04",
+				Suites:     []string{"jammy"},
+				Components: []string{"main", "universe"},
+				PubKeys:    []*packet.PublicKey{testKey.PubKey},
+			},
+		},
+		Packages: map[string]*setup.Package{
+			"mypkg": {
+				Archive: "ubuntu",
+				Name:    "mypkg",
+				Path:    "slices/mydir/mypkg.yaml",
+				Slices: map[string]*setup.Slice{
+					"myslice": {
+						Package: "mypkg",
+						Name:    "myslice",
+						Contents: map[string]setup.PathInfo{
+							"/path/**": {Kind: "generate", Generate: "foo"},
+						},
+					},
+				},
+			},
+		},
+	},
+	selslices: []setup.SliceKey{{"mypkg", "myslice"}},
+	selerror:  "slice mypkg_myslice has invalid 'generate' for path /path/\\*\\*: \"foo\", consider an update if available",
+}, {
+	summary: "Paths with generate: manifest must have trailing /**",
+	input: map[string]string{
+		"slices/mydir/mypkg.yaml": `
+			package: mypkg
+			slices:
+				myslice:
+					contents:
+						/path/: {generate: "manifest"}
+		`,
+	},
+	relerror: "slice mypkg_myslice path /path/ must end with /\\*\\* for 'generate: manifest' to be valid",
+}, {
+	summary: "Paths with generate: manifest must not have any other wildcard except the trailing **",
+	input: map[string]string{
+		"slices/mydir/mypkg.yaml": `
+			package: mypkg
+			slices:
+				myslice:
+					contents:
+						/pat*h/to/dir/**: {generate: "manifest"}
+		`,
+	},
+	relerror: "slice mypkg_myslice path /pat\\*h/to/dir/\\*\\* must not contain any other wildcard characters except trailing \\*\\* for 'generate: manifest' to be valid",
```

Comment:
Same as above, let's try to match the convention of "cannot ..." or "invalid ..." and make the message shorter if possible.